subsys/bluetooth/services/ots/ots_olcp.c

```diff
@@ -156,54 +156,36 @@ static enum bt_gatt_ots_olcp_res_code olcp_proc_execute(
 	}
 };
 
-static enum bt_gatt_ots_olcp_res_code olcp_command_decode(
-	const uint8_t *buf, struct bt_gatt_ots_olcp_proc *proc)
+static int olcp_command_decode(
+	const uint8_t *buf, uint16_t len,
+	struct bt_gatt_ots_olcp_proc *proc)
 {
 	memset(proc, 0, sizeof(*proc));
 
 	proc->type = *buf++;
+	len -= OLCP_PROC_TYPE_SIZE;
```

Comment:
What if `len < OLCP_PROC_TYPE_SIZE`?

Reply:
I opted to handle the scenario where the request length isn't large enough to even indicate an opcode outside of the `olcp_command_decode`. 

In `bt_gatt_ots_olcp_write` there is handling as such:
```c
	if (len < OLCP_PROC_TYPE_SIZE) {
		LOG_ERR("Invalid size of OLCP Write Request");
		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
	}
```

Analogous handling exists in `bt_gatt_ots_oacp_write`.

I am happy to move the `len < OLCP_PROC_TYPE_SIZE` inside of the `olcp_command_decode` function. This was just my preference. I also did this so in the `-EBADMSG` case handling inside of `bt_gatt_ots_olcp_write`, the opcode could be logged unconditionally as we are guaranteed that at least the opcode was present and was decoded.

I can also see the merit of locating this check inside of the `olcp_command_decode` function so all the validation is done in a single location but then we have to make a decision for logging of opcode based on the length of the request or add a new error code returned from `olcp_command_decode` to differentiate the two scenarios. The alternative is that we just do not log the opcode.

Reply:
Alright, that's alright, but please do consider adding an __ASSERT or something here then, otherwise it will underflow silently in case of an bug somewhere.